### ISSUE TITLE
Clarify docs for move's options 

### ISSUE THREAD

**Issue by valorloff** [jprichardson/node-fs-extra#348]
```
fs.move(src,dest,[true],function(error) {
            if (error) throw error;
            console.log('Moved to '+dest);
          });
```
Error: EEXIST: file already exists   .....

**Comment by RyanZim** [comment]
Docs should be clarified, correct usage of the overwrite option is: 
```js
fs.move(src,dest,{overwrite:true},function(error) {
  if (error) throw error;
  console.log('Moved to '+dest);
});
```
`options` must be an object.

Example: https://runkit.com/57ebc041d4a27f140028fd78/588a192238228e0015883bef.

**Comment by valorloff** [comment]
nice!
thanks!

**Comment by valorloff** [comment]
```
fs.move(oldFile, newFile,{owerwrite:true},(err)=>{
             if (err) throw err;
           });
```
```
Error: EEXIST: file already exists, link '/home/valaoffice/static/html/bla-bla.html' -> '/home/valaoffice/static/html/bla-bla-ma.html'
W20170127-11:18:11.649(2)? (STDERR)     at Error (native)
```
why?

**Comment by RyanZim** [comment]
@valorloff  `link` is part of the move operation. What is the value of `oldFile` & `newFile` when this error occurs?

**Comment by valorloff** [comment]
```
 if (err) throw err;   // 269
Error: EEXIST: file already exists, link '/home/valaoffice/static/html/k-t/s-p-200-m.html' -> '/home/valaoffice/static/html/d-m/s-p-200-m.html'
at Error (native)
```
moreover, moving file don't erase src file, and create new file only, this case above - exactly result of repeated fs.move

**Comment by RyanZim** [comment]
What is the value of `oldFile` & `newFile` when this error occurs? Without this info, we cannot proceed to debug this.

**Comment by valorloff** [comment]
console.log(src, dest);
src:  /home/valaoffice/static/html/aksessuary-dlya-mobilnyh-telefonov/Data-kabeli-adaptery/skotch-prozrachnyy-200-m.html 
dest:  /home/valaoffice/static/html/dlya-gazovoy-plity/skotch-prozrachnyy-200-m.html

**Comment by RyanZim** [comment]
@valorloff I cannot reproduce this error. Please create a repository that demonstrates this error and post it to GitHub. Open a new issue with a link to the repo. Thanks!

**Comment by valorloff** [comment]
I rewrite to:
```
if (!fs.existsSync(dest)) { // dest file not exist 
      if (!fs.existsSync(src)) {
      fs.writeFile(dest, SSR.render('productHtml'),
      function (err) {
        if (err) throw err;
        else  console.log('html file writed!');
      });
    } else {
      fs.rename(src,dest,(err)=>{
        if (err) throw err;
      })
    }
} else // dest file already exist
       	if (fs.existsSync(src)) {
       	  fs.rename(src,newHtmlFile,(err)=>{
          if (err) throw err;
        })
     }
```
and works perfectly for me, 
can it help you?

**Comment by lolmaus** [comment]
I'm running into the same issue. `{overwrite: true}` does not prevent this:

> Error: EEXIST: file already exists, link '/home/lolmaus/Code/lolma-us/tmp/deploy-dist/staticboot/.nojekyll' -> '/home/lolmaus/Code/lolma-us/tmp/deploy-dist'

**Comment by lolmaus** [comment]
Hmm, when this happens, the file `/home/lolmaus/Code/lolma-us/tmp/deploy-dist/.nojekyll` does not exist. O_o

**Comment by lolmaus** [comment]
I replaced `move` with `rename` and it worked fine.

**Comment by RyanZim** [comment]
Docs fixed in #384.
